prow/tide/tide_test.go

```diff
@@ -1532,6 +1539,76 @@ func TestSync(t *testing.T) {
 	}
 }
 
+func TestIsPassing(t *testing.T) {
+	yes := true
+	no := false
+	headSHA := "head"
+	success := string(githubql.StatusStateSuccess)
+	failure := string(githubql.StatusStateFailure)
+	testCases := []struct {
+		name             string
+		passing          bool
+		config           *config.Branch
+		combinedContexts map[string]string
+	}{
+		{
```

Comment:
Please add some test cases for optional contexts.

Reply:
done